### ISSUE TITLE
Spring No XML Example

### ISSUE THREAD

**Issue by mlaccetti** [martypitt/swagger-springmvc-example#17]
Not entirely sure if this belongs here or on the actual swagger-springmvc project; tried to follow the given steps to get Swagger working with SpringMVC in a no XML environment: servlet 3, @Configuration everywhere.  Ended up doing the following:
1. Add a `PropertyPlaceholderConfigurer` to the existing Spring configuration class:

``` java
@Bean
public static PropertyPlaceholderConfigurer swaggerProperties() {
    final PropertyPlaceholderConfigurer swaggerProperties = new PropertyPlaceholderConfigurer();
    swaggerProperties.setLocation(new ClassPathResource("swagger.properties"));
    return swaggerProperties;
}
```
1. Modify the `WebApplicationInitializer` to load the `DocumentationConfig`:

``` java
// Create the 'root' Spring application context
AnnotationConfigWebApplicationContext ctx = new AnnotationConfigWebApplicationContext();
ctx.register(SpringBaseConfiguration.class, SpringWebConfiguration.class, DocumentationConfig.class);
```

I didn't have to create any other beans or instantiate any other objects.

**Comment by dilipkrish** [comment]
@mlaccetti That is totally correct. Some of it has to do with transitioning existing applications with xml config to using the java config. Thanks for this info. I'll add it to the wiki.

**Comment by alanma** [comment]
@mlaccetti, @dilipkrish, 

I've tried the following but could not get the DocumentConfiguration bean registered and loaded:

public class WebApplicationInitializer extends AbstractAnnotationConfigDispatcherServletInitializer{

...
@Override
    protected Class<?>[] getRootConfigClasses() {
        return new Class<?>[] { RootConfiguration.class, DocumentationConfig.class };
    }
...
}

I got the following error, any idea how to work around it? Thanks.

Aug 11, 2013 9:59:03 PM org.apache.catalina.core.StandardContext listenerStart
SEVERE: Exception sending context initialized event to listener instance of class org.springframework.web.context.ContextLoaderListener
java.lang.ArrayStoreException: sun.reflect.annotation.TypeNotPresentExceptionProxy

**Comment by dilipkrish** [comment]
@alanma Have you checked the web.xml and the context file to see if it is setup correctly as well?

Take a look at [this file](https://github.com/martypitt/swagger-springmvc-example/blob/master/src/main/webapp/WEB-INF/spring-servlet.xml)

**Comment by alanma** [comment]
@dilipkrish, I can get the xml based configuration working as you did in the example project. I'm trying to use non-xml, 100% Java based config in my test project. However, I have issues when registering the DocumentationConfig. Wonder if you could try the Java based config? Thanks.

**Comment by mlaccetti** [comment]
@alanma I'll dig out my config tomorrow - I'm 100% annotations, no XML.

**Comment by mlaccetti** [comment]
@alanma @dilipkrish Here's a gist with a very very basic setup for Swagger using only annotations: https://gist.github.com/mlaccetti/6221808

**Comment by mlaccetti** [comment]
Quick update - since the gist may not have enough information, I've created a basic quickstart with a sample REST controller and Swagger/SpringMVC integrated: https://github.com/mlaccetti/swagger-springmvc-quickstart

**Comment by alanma** [comment]
@mlaccetti, thanks for the example project/gist. I've figured out my project issues reported earlier. It has been caused by my multi-module setup with multiple PropertyPlaceholderConfigurers. I'll upload my projects to github as soon as I get it all working. Thanks again Mike!

**Comment by dilipkrish** [comment]
:+1: Thanks @mlaccetti! Such problems are really hard to figure out because they can get so specific! @alanma glad its working now. 

Also I just released 0.6.0 version. Its got a ton of good fixes!

**Comment by targeter21** [comment]
@mlaccetti I just tried packaging and running the quickstart code and received the following message:

"Can't read swagger JSON from http://localhost:8080/swagger-springmvc-quickstart/api-docs/". Thoughts? My jetty is using 8080 by default and all the starting info should be correct...

**Comment by mlaccetti** [comment]
@targeter21 Sorry, not much use.  I updated the quickstart to use swagger-springmvc-0.8.3-SNAPSHOT and the most recent swagger-ui.  Seems to be working nicely, though I haven't tested generics support.

**Comment by targeter21** [comment]
Thanks! How a lot of this works is still very magical to me lol. I will definitely try this later. :+1:

**Comment by dilipkrish** [comment]
Sorry folks I'll go back and update the example to use 0.8.x ... For now this is the most recent example https://github.com/adrianbk/swagger-springmvc-demo

**Comment by alanma** [comment]
@targeter21, @sathishk has a forked version of @dilipkrish's swagger-springmvc-demo with full Java config. You might want to check it out:

https://github.com/sathishk/swagger-springmvc-demo
